doc/tags/types.rst

```diff
@@ -0,0 +1,30 @@
+``types``
+=======
+
+The ``types`` tag declares the types for one or more names variables.
+
+To do this, specify a :ref:`mapping <twig-expressions>` of names to their types as string constants. 
+
+Here is how you declare that variable ``foo`` is a boolean, while ``bar`` is an integer (see note below):
+
+.. code-block:: twig
+
+    {% types {
+        foo: 'bool',
+        bar: 'int',
+    } %}
+
+You may put these all on one line. However, using one line per variable improves readability.
+
+This tag has no effect on the template's output or runtime behavior.
+
+Its purpose is to enable designers and developers to document and specify the context's available
+and/or required variables.
```

Comment:
I would remove this sentence (see my comment below about creativity).

Reply:
I do want to point out the raison d'être for this to hopefully push adoption.

If the tag doesn't affect templates and seemingly has no point (if you're just looking at the official documentation), then won't many developers ignore and forget about this tag entirely?

I'm fine with removing this to keep the docs clean and to the point. I just wanted to bring up why I included this. Let me know what you prefer.

Reply:
I see. Maybe we could rephrase to suggest that this is one possible use case?

Reply:
How about

> While Twig itself does not validate variables or their types, this tag enables extensions to do this.

Reply:
That works for me